### PR TITLE
Remove correction to intercept

### PR DESCRIPTION
**Pull request by jyliuu**
The intercept should not be corrected here.

### CODE REVIEW

**Comment by jemus42** [comment]
But now the tests fail because the components no longer sum up to the prediction?

**Comment by mnwright** [comment]
This should be fixed but is slightly more complicated: The 0.5 is the base score of xgboost, which is the default value. With the defaults, this should be 0.5 for regression, but 0 for (binary) classification (because logit(0.5) = 0). 

The `base_score `is saved in `xg$params$base_score`, but only if it is manually set. If not, we can probably assume the default of 0.5. For `"binary:logistic"` loss, we have to use `logit(base_score)` here. 

Have to check other loss functions and multiclass.

**Review by mnwright — CHANGES_REQUESTED** [review]
See comment.

**Comment by jemus42** [comment]
I extended the basic xgboost test checking for the components summing up to the prediction to binary classification just in case. 
Considered doing the same for multiclass and realized how that's a whole can of worms I'm choosing to kick down the road for now 🥴

**Comment by jyliuu** [comment]
Okay I will close this PR, I thought this was due to a discussion which I perhaps misremembered.

**Comment by jemus42** [comment]
If I understand correctly, the issue is not that the .5 is wrong here, but rather that the .5 is only sometimes correct and in other cases needs to be adjusted.

Unfortunately I can't seem to be able to figure out what objective xgb used from a model object after training unless explicitly set, so that makes this somewhat hard to do correctly.
In any case, I think maybe this should be an open issue.
